File: data/scripts/sseq.py
#http://rosalind.info/problems/sseq/

def lettura(filename):
    data = []
    with open(filename) as f:
        record = ""
        first = True
        for riga in f:
            riga = riga.rstrip("\n")
            if riga[0] == ">":
                if not first:
                    data.append(record)
                record = ""
                first = False
            else:
                record += riga
    data.append(record)
    return data

def main():
    seqs = lettura("dataset.txt")
    s, t = seqs[0], seqs[1]
    posizioni = []
    j = 0
    for i in range(len(s)):
        if j < len(t) and s[i] == t[j]:
            posizioni.append(i + 1)
            j += 1
    print(" ".join(str(p) for p in posizioni))

if __name__ == "__main__":
    # execute only if run as a script
    main()

File: public/js-solutions/sseq.mjs
// Finding a Spliced Motif (Rosalind ID: SSEQ) - soluzione JavaScript
// indipendente, non una trascrizione di problem.py: stessa logica
// (parsing FASTA multi-record, poi ricerca greedy delle posizioni in s
// dove t compare come sottosequenza), riscritta in modo idiomatico per
// JS.
//
// Contratto: riceve il contenuto testuale di dataset.txt, restituisce
// l'output testuale (stessa forma dell'output Python).
function parseFasta(testo) {
  const righe = testo.split("\n").map((r) => r.replace(/\r$/, ""));
  const data = [];
  let record = "";
  let first = true;
  for (const riga of righe) {
    if (riga.startsWith(">")) {
      if (!first) data.push(record);
      record = "";
      first = false;
    } else {
      record += riga;
    }
  }
  data.push(record);
  return data;
}

export default function solve(datasetText) {
  const seqs = parseFasta(datasetText);
  const [s, t] = seqs;

  if (!s || !t) {
    throw new Error("Input non valido: attesi almeno due record FASTA");
  }

  const posizioni = [];
  let j = 0;
  for (let i = 0; i < s.length && j < t.length; i++) {
    if (s[i] === t[j]) {
      posizioni.push(i + 1);
      j++;
    }
  }

  return `${posizioni.join(" ")}\n`;
}
